general/src/types/annotated.rs

```diff
@@ -332,14 +322,36 @@ impl<T: ToValue> Annotated<T> {
             true
         }
     }
+
+    /// Modifies this value based on the action returned by `f`.
+    #[inline]
+    pub fn apply<F, R>(&mut self, f: F)
+    where
+        F: FnOnce(&mut T, &mut Meta) -> R,
+        R: Into<ValueAction>,
+    {
+        let result = match (self.0.as_mut(), &mut self.1) {
+            (Some(value), meta) => f(value, meta).into(),
+            (None, _) => Default::default(),
+        };
+
+        match result {
+            ValueAction::Discard => self.0 = None,
+            ValueAction::Keep => (),
+            ValueAction::MoveIntoOriginalValue => {
+                self.1
+                    .set_original_value(self.0.take().map(ToValue::to_value));
```

Comment:
I would make it a hard rule that `process_*` must not modify the value unless it wants that value to end up in `original_value`. Otherwise a clone would be necessary.

Reply:
Looked through the code again, we will not have this case. It might be good to add a sentence about this in the doc comment for `DeleteSoft`.

Reply:
Resolved in meatspace